azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/transport/mqtt/mqtt_transport.py

```diff
@@ -245,6 +244,48 @@ def _get_telemetry_topic(self):
         topic += "/messages/events/"
         return topic
 
+    def _encode_properties(self, message_to_send, topic):
+        system_properties = dict()
+        if message_to_send.message_id:
+            system_properties["$.mid"] = message_to_send.message_id
+
+        if message_to_send.correlation_id:
+            system_properties["$.cid"] = message_to_send.correlation_id
+
+        if message_to_send.user_id:
+            system_properties["$.uid"] = message_to_send.user_id
+
+        if message_to_send.to:
+            system_properties["$.to"] = message_to_send.to
+
+        if message_to_send.content_type:
+            system_properties["$.ct"] = message_to_send.content_type
+
+        if message_to_send.content_encoding:
+            system_properties["$.ce"] = message_to_send.content_encoding
+
+        if message_to_send.expiry_time_utc:
+            system_properties["$.exp"] = (
+                message_to_send.expiry_time_utc.isoformat()
+                if isinstance(message_to_send.expiry_time_utc, date)
+                else message_to_send.expiry_time_utc
+            )
+
+        # Question TODO : haven't encoded these properties here, i see it encoded in JAVA
```

Comment:
>Question TODO [](start = 10, length = 13)

unless you intend to solve this before checking in, please have a work item on the task board to track this down.

Reply:
[https://msazure.visualstudio.com/One/_workitems/edit/3605257](https://msazure.visualstudio.com/One/_workitems/edit/3605257)

---
In reply to: [240413057](https://github.com/Azure/azure-iot-sdk-python-preview/pull/34#discussion_r240413057) [](ancestors = 240413057)